### ISSUE TITLE
Sudden drop in return

### ISSUE THREAD

**Issue by MoreanP** [PKU-AI-Edge/CORRO#1]
![image](https://user-images.githubusercontent.com/37109784/197191345-2848797f-1670-4054-849b-89151aa82f06.png)
Hello. 
When I reproduce the paper, I use the point-robot-v1 environment and find that the return converges and then drops suddenly。
But I didn't modify the code。
What reason is this？

**Comment by Folly135** [comment]
Hi! I would like to know if you have solved this problem and does this image of yours corresponds to the RETURN image of the TRAIN stage? I don't seem to be able to draw the part corresponding to Figure 2 according to the Readme provided by the author, and the subsequent OOD should correspond to the contents of Table 1.

**Comment by MoreanP** [comment]
> Hi! I would like to know if you have solved this problem and does this image of yours corresponds to the RETURN image of the TRAIN stage? I don't seem to be able to draw the part corresponding to Figure 2 according to the Readme provided by the author, and the subsequent OOD should correspond to the contents of Table 1.

Yes, this graph corresponds to Figure 2 in the paper, which is a test return using iid data during training and was drawn using Tensorboard. The author did not provide the code for drawing the original image of the paper. I didn't solve the problem of sudden performance degradation in this image later.

**Comment by Folly135** [comment]
> > 你好！我想知道你是否解决了这个问题，你的这张图片是否对应于TRAIN阶段的RETURN图片？我根据作者提供的Readme似乎无法画出图2对应的部分，后续的OOD应该对应表1的内容。
> 
> 是的，这个图对应论文中的图2，是训练时使用iid数据的测试返回，是使用Tensorboard绘制的。作者没有提供绘制论文原图的代码。我后来并没有解决这个镜像中性能突然下降的问题。

Thank you very much for your reply, this part corresponds to the Offline Meta-RL section inside the Readme right?

**Comment by Folly135** [comment]
> > Hi! I would like to know if you have solved this problem and does this image of yours corresponds to the RETURN image of the TRAIN stage? I don't seem to be able to draw the part corresponding to Figure 2 according to the Readme provided by the author, and the subsequent OOD should correspond to the contents of Table 1.
> 
> Yes, this graph corresponds to Figure 2 in the paper, which is a test return using iid data during training and was drawn using Tensorboard. The author did not provide the code for drawing the original image of the paper. I didn't solve the problem of sudden performance degradation in this image later.

Thank you very much for your reply, this part corresponds to the Offline Meta-RL section inside the Readme right?

**Comment by MoreanP** [comment]
> > > Hi! I would like to know if you have solved this problem and does this image of yours corresponds to the RETURN image of the TRAIN stage? I don't seem to be able to draw the part corresponding to Figure 2 according to the Readme provided by the author, and the subsequent OOD should correspond to the contents of Table 1.
> > 
> > 
> > Yes, this graph corresponds to Figure 2 in the paper, which is a test return using iid data during training and was drawn using Tensorboard. The author did not provide the code for drawing the original image of the paper. I didn't solve the problem of sudden performance degradation in this image later.
> 
> Thank you very much for your reply, this part corresponds to the Offline Meta-RL section inside the Readme right?

yes

**Comment by nanzhu2003** [comment]
Hello, I'm trying running the code now but got some problems. Can I ask you which model did you load when run the test_ood_context.py？When I run this code, it always showed that the input size is incompatible with the parameter that I trained through the train_offpolicy_with_trained_encoder. Thank you a lot!

**Comment by MoreanP** [comment]
> Hello, I'm trying running the code now but got some problems. Can I ask you which model did you load when run the test_ood_context.py？When I run this code, it always showed that the input size is incompatible with the parameter that I trained through the train_offpolicy_with_trained_encoder. Thank you a lot!

It's a long time after did this, I don't remember very well. Which model reported an error?

**Comment by nanzhu2003** [comment]
> > Hello, I'm trying running the code now but got some problems. Can I ask you which model did you load when run the test_ood_context.py？When I run this code, it always showed that the input size is incompatible with the parameter that I trained through the train_offpolicy_with_trained_encoder. Thank you a lot!
> 
> It's a long time after did this, I don't remember very well. Which model reported an error?
Not exactly a problem. I finally found that it's because that I train the model with datasets I collected by myself so I don't have the behavior policy model. Now I have trained it. Anyhow, thanks a lot for your reply!
